tracer/src/Datadog.Trace/IAST/IastRequestContext.cs

```diff
@@ -41,6 +45,52 @@ internal void AddVulnerability(Vulnerability vulnerability)
         }
     }
 
+    internal void AddRequestBody(object body)
+    {
+        try
+        {
+            var bodyExtracted = ObjectExtractor.Extract(body);
```

Comment:
Although it's been optimized the ObjectExtractor is quite an expensive call, I wonder if there's some way we share the resulting object if it's being sent to the WAF as well.

Reply:
Yes, it can be done. I have updated the code. Thank you for the hint!